ShopAPI/ShopAPI/Controllers/ShopController.cs

```diff
@@ -119,5 +121,60 @@ public async Task<IActionResult> DeleteProduct(int id)
 
             return NoContent();
         }
+
+        [HttpPost]
+        [Route("AddItemToCart/{cartId}")]
+        public async Task<ActionResult<CartItemDTO>> AddItemToCart([FromRoute] int cartId, [FromBody] AddItemDTO item)
+        {
+            Product? product = await _productService.GetProductAsync(item.Id);
+
+            if (product is null)
+            {
+                return NotFound();
+            }
+
+            await _cartService.AddItemAsync(cartId, item.Id, item.Quantity);
+
+            ProductDTO productDTO = product.ModelToDTO();
+            CartItemDTO cartItemDTO = new(productDTO, item.Quantity);
+
+            return Ok(cartItemDTO);
+        }
+
+        [HttpGet]
+        [Route("GetCart/{cartId}")]
+        public async Task<ActionResult<CartDTO>> GetCart(int cartId)
+        {
+            List<CartItem> cartItems = await _cartService.getCartItemsAsync(cartId);
+            var totals = Calculate.Totals(cartItems);
+
+            // Convert cartItems to cartItemsDTOs
+            List<CartItemDTO> cartItemDTOs = new();
+            foreach (var cartItem in cartItems)
+            {
+                CartItemDTO cartItemDTO = new(cartItem.Product.ModelToDTO(), cartItem.Quantity);
+                cartItemDTOs.Add(cartItemDTO);
+            }
```

Comment:
Interesting article I read recently that might apply here - https://www.code4it.dev/csharptips/initialize-collection-size/

Up to you if you though if you want to make the change

Reply:
Fixed in https://github.com/evan-scales/PalmetoProgrammers547/pull/10/commits/4030628e32dc8d38ed6f61ad652d4eda96a3185b